app/controllers/application_controller/filter/expression.rb

```diff
@@ -446,16 +446,16 @@ def update_from_exp_tree(exp)
 
       if val1 && [:datetime, :date].include?(val1[:type])     # Change datetime and date field values into arrays while editing
         self.exp_value = Array.wrap(exp_value)                # Turn date/time values into an array
-        val1[:date_format] = exp_value.to_s.first.include?('/') ? 's' : 'r'
+        val1[:date_format] = exp_value.first.to_s.include?('/') ? 's' : 'r'
         if key == EXP_FROM && val1[:date_format] == 'r'
-          val1[:through_choices] = Expression.through_choices(exp_value[0])
+          val1[:through_choices] = exp_value.empty? ? nil : Expression.through_choices(exp_value[0])
```

Comment:
sorry to extend this across another repo but...

if `Expression.through_choices` accepted a nil and returned a nil, that could simplify this

Reply:
I will make this change, and happily, but just for reference this is not going to extend across another repo: it lives https://github.com/ManageIQ/manageiq-ui-classic/blob/16406cb713e2646c5928efb287df5c24ec768b39/app/controllers/application_controller/filter/expression.rb#L585